### ISSUE TITLE
brackets-memory-db update function bad entity access

### ISSUE THREAD

**Issue by ghost** [Drarig29/brackets-storage#12]
Hi @Drarig29 and @Tandashi,

I really appreciate your work on the brackets-storage and the brackets-manager in general. I am using the brackets-memory-db and run into a serious issue the other day.
The update method of the brackets-storage is accessing the entities of the tables by the id as the array index.
This is https://github.com/Drarig29/brackets-storage/blob/2d86211a6372368f08343b5db014de8ce11f647e/brackets-memory-db/index.ts#L193 https://github.com/Drarig29/brackets-storage/blob/2d86211a6372368f08343b5db014de8ce11f647e/brackets-memory-db/index.ts#L213 and https://github.com/Drarig29/brackets-storage/blob/2d86211a6372368f08343b5db014de8ce11f647e/brackets-memory-db/index.ts#L224.
If entities get deleted on the storage this is not anymore a correct behavior because the ids are not starting anymore with index 0.

The fix is straight-forward and I could provide a PR if you like.

Thanks

**Comment by Drarig29** [comment]
Thanks for the suggestion, PRs are welcome

**Comment by ghost** [comment]
Ok, I guess you have to allow me to setup a branch then.

**Comment by Tandashi** [comment]
> Ok, I guess you have to allow me to setup a branch then.

You don't need permission. Simply fork the repository. Create a branch in your own fork and than create a Pull Request in this repository.

See: https://git-scm.com/book/en/v2/GitHub-Contributing-to-a-Project

**Comment by Tandashi** [comment]
Do you have a minimal example where this issue occurs? So we can verify that the Pull Request indeed fixes the issue?

**Comment by Tandashi** [comment]
Replication Example:
```typescript
import { InMemoryDatabase } from "brackets-memory-db";
const storage = new InMemoryDatabase();

async function main() {
  const id = await storage.insert("stage", { first: true });
  const id2 = await storage.insert("stage", { second: true });

  console.log(await storage.select("stage")); // [ { id: 0, first: true }, { id: 1, second: true } ]

  await storage.delete("stage", { id });
  await storage.update("stage", id2, { second: true, updated: 1 });

  console.log(await storage.select("stage")); // [ { id: 1, second: true }, { second: true, updated: 1 } ]
}
main();
```

The last log should have only contained the second entry. The first should not exist but it does so since the indices have moved due to the deletion.

**Comment by Drarig29** [comment]
Fixed by https://github.com/Drarig29/brackets-storage/pull/13, and released in [brackets-memory-db@1.0.5](https://www.npmjs.com/package/brackets-memory-db/v/1.0.5)!

Thanks for the PR 😉
